cppcheck_junit.py

```diff
@@ -46,6 +46,8 @@ def parse_arguments() -> argparse.Namespace:
     )
     parser.add_argument("input_file", type=str, help="Cppcheck XML version 2 stderr file.")
     parser.add_argument("output_file", type=str, help="JUnit XML output file.")
+    parser.add_argument("error_exitcode", type=int, nargs="?", const=0, help="If errors are found, \
+                            integer <n> is returned instead of default 0.")
```

Comment:
Failure is defined as `1` (see `Exitstatus.failure`) so I think this help text is incorrect.

`1` is pretty universally considered the standard failure code. What is the situation/CI context where you need something other than `1`?

Reply:
This is based on the [--error-exitcode](https://www.mankier.com/1/cppcheck#--error-exitcode) option of cppcheck.  
In the context of CI and scripting someone might want to use a custom error code for a linter violation, so the standard `1` is reserved for failures in the script. This way they can be differentiated.

Reply:
Sounds good. I like matching `cppcheck`.

Shouldn't the help text still say `If errors are found, integer <n> is returned instead of default 1.`? Even better would be to inject `Exitstatus.failure` into this message and make it a format string.

Reply:
Yeah, let's use `Exitstatus.success` for consistency.

And yeah, thanks for the sanity check it's been a while since I looked at this code 😄. Glad to see someone is getting use out of it.

Reply:
Made the code adjustments and squashed the fixup commits to tidy things up.